File: HarvesterBack/Harvester/Harvester.Domain/Models/Order.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Harvester.Domain.Models.Enums;

namespace Harvester.Domain.Models
{
    public class Order
    {
        public int Id { get; set; }

        public int FieldId { get; set; }
        public Field? Field { get; set; }

        public int CombineId { get; set; }
        public Combine? Combine { get; set; }

        public DateTime OrderDate { get; set; }
        public DateTime ScheduledDate { get; set; }
        public OrderStatus Status { get; set; }
        public StrawProcessingMethod StrawProcessingMethod { get; set; }

        public int EstimatedTime { get; set; }
        public decimal EstimatedPrice { get; set; }
        public decimal TotalPrice { get; set; }
        public ICollection<Payment>? Payments { get; set; }
        public bool IsDeleted { get; set; }

        public bool IsArchived { get; set; }
    }
}

File: HarvesterFront/src/app/core/models/Order.ts
import { Combine } from './Combine';
import { OrderStatus } from './enums/order-status';
import { StrawProcessingMethod } from './enums/straw-processing-method';
import { Field } from './Field';

export interface Order {
  id: number;
  field: Field;
  combine: Combine;
  orderDate: Date;
  scheduledDate: Date;
  status: OrderStatus;
  strawProcessingMethod: StrawProcessingMethod;
  estimatedTime: number;
  estimatedPrice: number;
  totalPrice: number;
  isArchived: boolean;
}
